unison-core/src/Unison/Name.hs

```diff
@@ -574,28 +575,35 @@ unsafeFromString =
 --
 -- Performs very minor validation (a name can't be empty, nor contain a '#' character [at least currently?]) but makes
 -- no attempt at rejecting bogus names like "foo...bar...baz".
-unsafeFromText :: HasCallStack => Text -> Name
-unsafeFromText = \case
-  "" -> error "empty name"
-  "." -> Name Relative ("." :| [])
-  ".." -> Name Absolute ("." :| [])
+fromText :: Text -> Maybe Name
```

Comment:
Add a safe `fromText` which was missing, implement the unsafe variant in terms of the safe one.